app/src/hooks/useGetCounter.tsx

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from 'react'
+
+export default function useGetCounter(seconds: number, isCountingUp: boolean) {
+  const [counterContent, setCounterContent] = useState(isCountingUp ? 0 : seconds)
+  const [isPaused, setIsPaused] = useState(false)
+
+  const handlePause = () => {
+    setIsPaused((isPaused) => !isPaused)
+  }
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      if (isPaused) return
+
+      setCounterContent((prevCounter) => {
+        if (isCountingUp) {
+          return prevCounter < seconds ? prevCounter + 1 : seconds
+        }
+
+        return prevCounter > 0 ? prevCounter - 1 : 0
```

Comment:
I think this calculation and the previous one could be moved outside the component so it's more unit-testable.